packages/@react-spectrum/textfield/src/TextFieldBase.tsx

```diff
@@ -54,13 +54,17 @@ function TextFieldBase(props: TextFieldBaseProps, ref: Ref<TextFieldRef>) {
     inputRef,
     isLoading,
     loadingIndicator,
+    value,
+    defaultValue,
     validationIconClassName
   } = props;
   let {hoverProps, isHovered} = useHover({isDisabled});
   let domRef = useRef<HTMLDivElement>(null);
   let defaultInputRef = useRef<HTMLInputElement | HTMLTextAreaElement>(null);
   inputRef = inputRef || defaultInputRef;
 
+  let inputValue = (value || value === '') ? value : defaultValue;
```

Comment:
I think you just want this
```
value ?? defaultValue ?? '';
```
and lets name the variable readOnlyText instead of inputValue, currently it sounds like it's going to be used to control the input state

found looking at `/story/textfield--isreadonly-true`